Behavior/Announce.cs

```diff
@@ -0,0 +1,131 @@
+            string channelDescription)
+        {
+            ValidateArg.IsNotNullOrWhiteSpace(channelName, nameof(channelName));
+
+            await Announce.SendAnnouncementMessageAsync(
+                guildConnection: requestAuthor.Guild,
+                guild: guild,
+                message: $"<@{requestAuthor.Id}> created a new channel named **{channelName}** with the description \"{channelDescription}\"\n" +
+                         $"Let me know if you're interested by using a command like, \"/join {channelName}\"")
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Annoucees the rename of a channel in the given guild if an annoucement channel is configured.
+        /// </summary>
+        /// <param name="guild">Information about the guild. May not be null.</param>
+        /// <param name="requestAuthor">The author of the channel rename request. May not be null.</param>
+        /// <param name="oldChannelName">The old name of the channel that was renamed. May not be null.</param>
+        /// <param name="newChannelName">The new name of the channel that was renamed. May not be null.</param>
+        /// <returns>A task indicating completion.</returns>
+        public static async Task ChannelRename(
+            Guild guild,
+            SocketGuildUser requestAuthor,
+            string oldChannelName,
+            string newChannelName)
+        {
+            ValidateArg.IsNotNullOrWhiteSpace(oldChannelName, nameof(oldChannelName));
+            ValidateArg.IsNotNullOrWhiteSpace(newChannelName, nameof(newChannelName));
+
+            await Announce.SendAnnouncementMessageAsync(
+                guildConnection: requestAuthor.Guild,
+                guild: guild,
+                message: $"<@{requestAuthor.Id}> changed the name of the **{oldChannelName}** channel to **{newChannelName}**.")
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Announces a change to the description of a channel in the given guild if an annoucement channel is configured.
+        /// </summary>
+        /// <param name="guild">Information about the guild. May not be null.</param>
+        /// <param name="requestAuthor">The author of the channel redescribe request. May not be null.</param>
+        /// <param name="channelName">The name of the channel being described. May not be null.</param>
+        /// <param name="oldDescription">The old description of the channel.</param>
+        /// <param name="newDescription">The new description of the channel.</param>
+        /// <returns>A task indicating completion.</returns>
+        public static async Task ChannelRedescribe(
+            Guild guild,
+            SocketGuildUser requestAuthor,
+            string channelName,
+            string oldDescription,
+            string newDescription)
+        {
+            ValidateArg.IsNotNullOrWhiteSpace(channelName, nameof(channelName));
+
```

Comment:
Is it worth announcing if description goes from nothing to nothing?

Reply:
The code that calls into this should not request the announcement if the description didn't change. I'll add a check for that and also updated the message to account for either the old or new description being null or empty.